core/PRP/Geometry/plDrawableSpans.cpp

```diff
@@ -528,10 +528,16 @@ void plDrawableSpans::calcBounds()
         hsBounds3Ext world;
 
         world.setFlags(hsBounds3Ext::kAxisAligned);
+        hsVector3* localPoints = new hsVector3[verts.size()];
+        hsVector3* worldPoints = new hsVector3[verts.size()];
```

Comment:
Instead of manual memory management, consider:

```c++
std::vector<hsVector3> localPoints(verts.size());
```

Or even:

```c++
auto localPoints = std::make_unique<hsVector3[]>(verts.size());
```

Reply:
I swear, if only the courses I attended focused more on usage of the STD lib instead of constantly bashing it as "inefficient", I wouldn't have so many bad coding practices !

Doesn't build on Linux though, I'm looking into it...

Reply:
Not sure about your experience, but when I was in school it was mostly true, to my recollection.  Simultaneously incomplete and unoptimized in many cases, we used it for user-facing io but not much else.  It was stagnant for seemingly quite a while, too.

Reply:
I believe compilation can be achieved on Linux by adding `#include <memory>` in the files using `std::make_unique`.

Reply:
> Not sure about your experience, but when I was in school it was mostly true, to my recollection.

Gotta admit, C++11 wasn't widely used or supported back when I first learnt it using online tutorials. A few years later I had proper courses at school, but most students never actually programmed before that, so teachers spent most of their time explaining what pointers are. That was probably around the release of C++14, and I didn't have much use for C++ after that, so yeah.

Reply:
C++11 was in the distant future when I was in school. ;)